BreakinIn/Messages/ReptIn.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BreakinIn.Messages
+{
+    public class ReptIn : AbstractMessage
+    {
+        public override string _Name { get => "rept"; }
+
+        public string PERS { get; set; }
+        public string PROD { get; set; }
+        public string LANG { get; set; }
+
+        public override void Process(AbstractEAServer context, EAClient client)
+        {
+            var mc = context as MatchmakerServer;
+            if (mc == null) return;
+
+            var PlayerReportingPlayer = new ReptOut()
+            {
+                PERS = PERS,
+            };
+
+            var user = client.User;
+            if (user == null) return;
+
+
+
+            if (user.PersonaName == PERS)
+            {
+                Console.WriteLine(user.PersonaName + " attempted to report themselves, which is not allowed.");
+                client.SendMessage(new ReportingSelf());
+            }
+            else
+            {
+                if (user.CurrentRoom.Name.ToString() == null)
+                {
+                    Console.WriteLine(user.PersonaName + " has reported " + PERS + ". Both players seem to be in a certain room, but it doesn't have a name.");
```

Comment:
You haven't tried to get the target user, let alone check what room they are in, so this log does not make sense. If you changed the above to be `user.CurrentRoom != null`, you could say that `user.PersonaName` has reported `PERS`, but they are not in a room. (this is an invalid state, as you shouldn't be able to arbitrarily report someone you aren't playing with.

Reply:
I was thinking of a what if situation such as the reported player behaving inappropriately, but then leaving while the player making the report is still on the "Do you want to report <playername>?" window. If the player were to then still file the report, then they would not be able to as the other player is no longer in the room...

One good thing however, is that you can't just file reports aimlessly at everyone, the only time you can report someone is if both they and you are in the Matchmaker/Chatroom lobby.